File: 1127_DataCouch_AzureOpenAI_S6/src/Go/palindrome.go
// palindrome.go
package main

import (
	"fmt"
)

func isPalindrome(s string) bool {
	if s == "" {
		return false
	}

	left := 0
	right := len(s) - 1

	for left < right {
		fmt.Printf("Current elements left index: '%c' and right index: '%c'\n", s[left], s[right])

		if s[left] != s[right] {
			fmt.Println("The elements aren't the same, hence we return false")
			return false
		}

		left++
		right--
	}

	return true
}

File: 1206_DataCouch_AzureOpenAI_S7/src/CS/Tpp.Algo/Palindrome.cs
// Palindrome.cs

namespace Tpp.Algo;

public static class PalindromeChecker
{
    public static bool IsPalindrome(string s)
    {
        if (s == null)
        {
            return false;
        }

        int left = 0;
        int right = s.Length - 1;

        while (left < right)
        {
            WriteLine($"Current elements left index: '{s[left]}' and right index: '{s[right]}'");

            if (s[left] != s[right])
            {
                WriteLine("The elements aren't the same, hence we return false");
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}
